### PR TITLE
Update targetSdk version to 34

### PR DESCRIPTION
**Pull request by ashiagr**
## Description
This updates target SDK version to 34 (Android 14).

Closes https://github.com/Automattic/pocket-casts-android/issues/1569

## Testing Instructions

Smoke test the app

I've tested the following features:

 ✅ Episode playback (Fixed warning in https://github.com/Automattic/pocket-casts-android/pull/1754)
 ✅ OPML Import/ Export (Fixed in https://github.com/Automattic/pocket-casts-android/pull/1684)
 ✅ Download episode (Fixed in https://github.com/Automattic/pocket-casts-android/pull/1684)
 ✅ Cast (Fixed in https://github.com/Automattic/pocket-casts-android/pull/1686)
 ✅ Upload file
 ✅ Share
 ✅ Widgets, Shortcuts
 ✅ Login (Email, Google Sign-In), Sign up
 ✅ Syncing
 ✅ Playback from notification
 ✅ Play/ pause/ skip Bluetooth commands
 ✅ Subscriptions
 ✅ New episode notifications
 ✅  Runtime-registered broadcasts receivers 
 ✅ Add bookmark using headphones (Fixed in [7355d41](https://github.com/Automattic/pocket-casts-android/pull/2279/commits/7355d41f2bf662bf38bdfef0b5d657c428f7c97f))
 ✅ Smoke test release builds: phone, watch, automotive

## Checklist
- [ ] If this is a user-facing change, I have added an entry in CHANGELOG.md
- [x] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [ ] I have considered whether it makes sense to add tests for my changes
- [ ] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [ ] Any jetpack compose components I added or changed are covered by compose previews
- [ ] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.
 
#### I have tested any UI changes...

- [ ] with different themes
- [ ] with a landscape orientation
- [ ] with the device set to have a large display and font size
- [ ] for accessibility with TalkBack

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -241,7 +241,7 @@ auth = "com.google.android.gms:play-services-auth:20.4.0"
 barista = "com.adevinta.android:barista:4.2.0"
 browser = "androidx.browser:browser:1.4.0"
 browser-helper = "com.google.androidbrowserhelper:androidbrowserhelper:2.3.0"
-car = "androidx.car.app:app:1.2.0"
+car = "androidx.car.app:app:1.4.0-rc02"
```
Comment (ashiagr) [inline gradle/libs.versions.toml:244]: Updated to fix this error on creating a bookmark:

Screenshot 2024-05-29 at 3 40 25 PM

**Comment by MiSikora** [comment]
I'll wait with review for #2283 because changing target SDK might cause some new Lint issues to pop-up so we might need to fix them or create new baseline profiles.

**Review by MiSikora — APPROVED** [review]
I tested major areas of the app mentioned in this PR and in general played with the app. It all worked as expected. Nice work!

**Pull request merged**